notify/factory.go

```diff
@@ -27,57 +31,146 @@ import (
 	"github.com/grafana/alerting/receivers/wecom"
 )
 
-var receiverFactories = map[string]func(receivers.FactoryConfig) (NotificationChannel, error){
-	"prometheus-alertmanager": wrap(alertmanager.New),
-	"dingding":                wrap(dinding.New),
-	"discord":                 wrap(discord.New),
-	"email":                   wrap(email.New),
-	"googlechat":              wrap(googlechat.New),
-	"kafka":                   wrap(kafka.New),
-	"line":                    wrap(line.New),
-	"opsgenie":                wrap(opsgenie.New),
-	"pagerduty":               wrap(pagerduty.New),
-	"pushover":                wrap(pushover.New),
-	"sensugo":                 wrap(sensugo.New),
-	"slack":                   wrap(slack.New),
-	"teams":                   wrap(teams.New),
-	"telegram":                wrap(telegram.New),
-	"threema":                 wrap(threema.New),
-	"victorops":               wrap(victorops.New),
-	"webhook":                 wrap(webhook.New),
-	"wecom":                   wrap(wecom.New),
-	"webex":                   wrap(webex.New),
-}
-
-type NotificationChannel interface {
-	notify.Notifier
-	notify.ResolvedSender
-}
+// BuildReceiverIntegrations builds notifiers of the receiver and wraps each of them in Integration.
```

Comment:
This comment could be more helpful.

```suggestion
// BuildReceiverIntegrations creates integrations for each configured notification channel in GrafanaReceiverConfig.
// It returns a slice of Integration objects, one for each notification channel, along with any errors that occurred.
```